src/Agent.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -329,16 +331,18 @@ public AgentSettings LoadSettings()
             bool runAsService = command.GetRunAsService();
             if (runAsService)
             {
-                if (!new WindowsPrincipal(WindowsIdentity.GetCurrent()).IsInRole(WindowsBuiltInRole.Administrator))
-                {
-                    Trace.Error("Needs Administrator privileges for configure agent as windows service.");
-                    throw new SecurityException(StringUtil.Loc("NeedAdminForConfigAgentWinService"));
-                }
-
                 Trace.Info("Configuring to run the agent as service");
                 var serviceControlManager = HostContext.GetService<IWindowsServiceControlManager>();
                 serviceControlManager.ConfigureService(agentSettings, command);
             }
+            //This will be enabled with AutoLogon code changes are tested
+            else
+            {                
+                ConfigureAutoLogonIfNeeded(command);
```

Comment:
Exceptions are not thrown further as we decided to not error out if autologon config fails.

---
In reply to: [118601719](https://github.com/Microsoft/vsts-agent/pull/985#discussion_r118601719) [](ancestors = 118601719)